srml/babe/src/lib.rs

```diff
@@ -363,12 +364,20 @@ impl<T: Trait> Module<T> {
 	fn do_initialize() {
 		// since do_initialize can be called twice (if session module is present)
 		// => let's ensure that we only modify the storage once per block
-		let initialized = Self::initialized().unwrap_or(false);
+		let initialized = Self::initialized().is_some();
 		if initialized {
 			return;
 		}
 
-		Initialized::put(true);
+		#[derive(Default)]
+		struct Guard { maybe_vrf: MaybeVrf };
+		impl Drop for Guard {
+			fn drop(&mut self) {
+				Initialized::put(self.maybe_vrf.take())
+			}
+		}
+
+		let mut guard = Guard::default();
 		for digest in Self::get_inherent_digests()
```

Comment:
I know this code was already written like this but in practice this loop should only have one iteration so maybe we could rewrite this to make that more explicit? Using `find` instead of `filter_map` or just doing a `take(1)` at the end should work. We can probably avoid using this `Guard` if it's explicitly just getting one value from the iterator.